gcore/gdaldriver.cpp

```diff
@@ -949,6 +960,82 @@ GDALDataset *GDALDriver::CreateCopy( const char * pszFilename,
+        CPLXMLNode* xmlNode = CPLParseXMLString(pszOptionList);
+        for(CPLXMLNode* child = !xmlNode ? 0 : xmlNode->psChild ; child != 0 ; child = child->psNext)
+        {
+            if ((child->eType == CXT_Element) && EQUAL(child->pszValue, "Option"))
+            {
+                CPLXMLNode* nameAttribute = CPLGetXMLNode(child, "name");
+                if (nameAttribute != nullptr)
+                {
+                    const bool isInterleaveAttribute = nameAttribute->psChild &&
+                        (nameAttribute->psChild->eType == CXT_Text) &&
+                        EQUAL(nameAttribute->psChild->pszValue, "INTERLEAVE");
+                    if (isInterleaveAttribute)
+                    {
+                        for(CPLXMLNode* optionChild = child->psChild ; optionChild != 0 ; optionChild = optionChild->psNext)
+                        {
+                            if ((optionChild->eType == CXT_Element) && EQUAL(optionChild->pszValue, "Value"))
+                            {
+                                CPLXMLNode* optionChildValue = optionChild->psChild;
+                                if (optionChildValue && (optionChildValue->eType == CXT_Text))
+                                {
+                                    interleavesCSL = CSLAddString(interleavesCSL, optionChildValue->pszValue);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        CPLDestroyXMLNode(xmlNode);
+
+        const char* dstInterleaveBand =
+            (CSLFindString(interleavesCSL, "BAND") >= 0) ? "BAND" :
+            (CSLFindString(interleavesCSL, "BSQ") >= 0) ? "BSQ" :
+            nullptr;
+        const char* dstInterleaveLine =
+            (CSLFindString(interleavesCSL, "LINE") >= 0) ? "LINE" :
+            (CSLFindString(interleavesCSL, "BIL") >= 0) ? "BIL" :
+            nullptr;
+        const char* dstInterleavePixel =
+            (CSLFindString(interleavesCSL, "PIXEL") >= 0) ? "PIXEL" :
+            (CSLFindString(interleavesCSL, "BIP") >= 0) ? "BIP" :
+            nullptr;
+        const char* dstInterleave = !srcInterleave ? 0 :
+            EQUAL(srcInterleave, "BAND") ? dstInterleaveBand :
+            EQUAL(srcInterleave, "LINE") ? dstInterleaveLine :
+            EQUAL(srcInterleave, "PIXEL") ? dstInterleavePixel :
+            nullptr;
+        CSLDestroy(interleavesCSL);
+
+        char** dstInterleaveOptions = !dstInterleave ? 0 :
+          CSLSetNameValue(0, "INTERLEAVE", dstInterleave);
+        const bool bIsDstInterleaveSupported = !dstInterleaveOptions ? FALSE :
+          (GDALValidateCreationOptions(this, dstInterleaveOptions) == TRUE);
+        if (bIsDstInterleaveSupported)
+        {
+            papszOptionsWithInterleave = CSLDuplicate(papszOptions);
+            papszOptionsWithInterleave = CSLMerge(papszOptionsWithInterleave, dstInterleaveOptions);
```

Comment:
instead of calling CSLMerge(), you could just do
papszOptionsWithInterleave = CSLSetNameValue(papszOptionsWithInterleave, "INTERLEAVE", dstInterleave);